### PR TITLE
feat(bigquery): Add Query Acceleration to BigLake Examples

### PR DESCRIPTION
**Pull request by nevzheng**
## Description

+ Add missing connection_id to table in `bigquery_create_partitioned_biglake_table`
+ Add: Query Acceleration to examples

Fixes #<ISSUE-NUMBER>

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [ ] Yes, **merge** this PR after it is approved
- [x] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/bigquery/docs/create-cloud-storage-table-biglake

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by nevzheng** [comment]
+ @adhiggs for tech writer review

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 1]
```diff
@@ -90,6 +90,11 @@ resource "google_bigquery_dataset" "default" {
 # This creates a BigQuery table named "my_table" in the dataset "default".
 # The table has a single column named "column1", which is of type STRING
 # and is nullable.
+# The table has Automatic Metadata Caching enabled.
+# https://cloud.google.com/bigquery/docs/biglake-intro#metadata_caching_for_performance
+# `metadata_cache_mode` may be set to `MANUAL` and `max_staleness` ommitted to
```
Comment (wj-chen) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:95]: "omitted".

Also may be clearer if we said "Set `metadata_cache_mode` to `MANUAL` and omit `max_staleness` to use manual metadata refresh. Omit both `metadata_cache_mode` and `max_staleness` to disable metadata caching." here and in similar other descriptions below. But I'll defer the exact wording to @adhiggs.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 2]
```diff
@@ -101,14 +106,26 @@ resource "google_bigquery_table" "default" {
...
+    connection_id = google_bigquery_connection.default.name
     # This configures Hive partitioning for the BigQuery table,
     # partitioning the data by date and time.
     hive_partitioning_options {
       mode                     = "CUSTOM"
       source_uri_prefix        = "gs://${google_storage_bucket.default.name}/publish/{dt:STRING}/{hr:STRING}/{min:STRING}"
       require_partition_filter = false
     }
+    # This enables automatic metadata refresh.
+    # `MANUAL` for metadata refresh
```
Comment (wj-chen) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:118]: End a sentence with "." here and below.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf [hunk 3]
```diff
@@ -81,7 +81,13 @@ resource "google_bigquery_dataset" "default" {
...
 
 # This creates a BigQuery table named "my_table" in the dataset "default".
-# The table has three columns, named "country", "product", and "price", which are of types STRING, STRING, and INT64 respectively.
+# The table has three columns, named "country", "product", and "price", which
+# are of types STRING, STRING, and INT64 respectively.
+# The table has Automatic Metadata Caching enabled.
+# https://cloud.google.com/bigquery/docs/biglake-intro#metadata_caching_for_performance
+# `metadata_cache_mode` may be set to `MANUAL` and `max_staleness` ommitted to
+# use MANUAL Metadata refresh `metadata_cache_mode` and `max_staleness` may be
+# set to omitted to disable the feature.
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:90]: Suggestion: There are region tags are in this file. So, if you are using this sample in the cloud docs, kindly keep code comments to be simple, short, to the point.

**Comment by msampathkumar** [comment]
@nevzheng Going forward, for all commits and PR title, kindly following commits conventional format like `feat(bigquery): <short title>`

* feat - for new samples
* fix - fix any broken issues
* docs - add docs or code comments
* chore - clean and all other

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 1 above)
Comment (msampathkumar) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:95]: Just a suggestion. Use comments to explain details that are not so obviation or details that are not found in reference docs.

Usually users refer to [docs in Hashicorp](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table). So, avoid explaining sample details as in Hashicorp

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf [hunk 4]
```diff
@@ -81,7 +81,13 @@ resource "google_bigquery_dataset" "default" {
...
 resource "google_bigquery_table" "default" {
   depends_on = [time_sleep.wait_7_min]
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:92]: Move `depends_on` argument to the last. 

[Style Guide](https://developer.hashicorp.com/terraform/language/syntax/style) says `For blocks that contain both arguments and "meta-arguments" (as defined by the Terraform language semantics), list meta-arguments first and separate them from other arguments with one blank line. Place meta-argument blocks last and separate them from other blocks with one blank line.`

**Comment by msampathkumar** [comment]
@nevzheng PR looks good. Minors improvements can be made to the code comments & style. Waiting the user to address the reviewer suggestiions.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 1 above)
Comment (adhiggs) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:95]: Agree with msampathkumar; we can omit this sort of implementation detail. The important thing is that the code should run as little (preferably none) customization needed from the user.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 4 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:92]: done.

copied to all blocks and files.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 1 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:95]: done.
simplified

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 2 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:119]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 3 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:90]: done.

simplified

Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:90]: done.

**Review by wj-chen — APPROVED** [review]

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 5]
```diff
@@ -24,7 +24,9 @@
 */
 
 # [START bigquery_create_partitioned_biglake_table]
-# This creates a bucket in the US region named "my-bucket" with a pseudorandom suffix.
+
+# This creates a bucket in the US region named "my-bucket" with a pseudorandom
+# suffix.
 resource "random_id" "bucket_name_suffix" {
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:30]:
```suggestion
resource "random_id" "default" {
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf [hunk 6]
```diff
@@ -56,8 +56,9 @@ resource "google_project_iam_member" "default" {
 # This makes the script wait for seven minutes before proceeding.
 # This lets IAM permissions propagate.
 resource "time_sleep" "wait_7_min" {
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:58]:
```suggestion
resource "time_sleep" "default" {
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf [hunk 7]
```diff
@@ -99,7 +98,17 @@ resource "google_bigquery_table" "default" {
...
   }
+
+  # `max_staleness` is required for automatic metadata caching and must be
+  # specified as an interval literal.
+  # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#interval_literals
+  max_staleness = "0-0 0 10:0:0"
+
   deletion_protection = false
+
+  depends_on = [time_sleep.wait_7_min]
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:112]:
```suggestion
  depends_on = [time_sleep.default]
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 6 above)
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:58]:
```suggestion
resource "time_sleep" "default" {
```

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 8]
```diff
@@ -61,16 +63,16 @@ resource "google_project_iam_member" "default" {
   member  = "serviceAccount:${google_bigquery_connection.default.cloud_resource[0].service_account_id}"
 }
 
-# This makes the script wait for seven minutes before proceeding.
-# This lets IAM permissions propagate.
+# This makes the script wait for seven minutes before proceeding. This lets IAM
+# permissions propagate.
 resource "time_sleep" "wait_7_min" {
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:68]:
```suggestion
resource "time_sleep" "default" {
```

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 9]
```diff
@@ -46,8 +48,8 @@ resource "google_storage_bucket_object" "default" {
 # This queries the provider for project information.
 data "google_project" "project" {}
```
Comment (msampathkumar) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:49]:
```suggestion
data "google_project" "default" {}
```

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 5 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:30]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 9 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:49]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 8 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:68]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 6 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:58]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 7 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:112]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 10]
```diff
@@ -109,11 +110,20 @@ resource "google_bigquery_table" "default" {
...
       require_partition_filter = false
     }
+    # This enables automatic metadata refresh.
+    metadata_cache_mode = "AUTOMATIC"
   }
+
+
+  # `max_staleness` is required for automatic metadata caching and must be
+  # specified as an interval literal.
+  # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#interval_literals
```
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:120]:
```suggestion
  # This sets the maximum staleness of the metadata cache to 10 hours.
```
We can lean on the variable list on L23 if users want to get deeper into the technicalities of this.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf [hunk 11]
```diff
@@ -80,10 +81,8 @@ resource "google_bigquery_dataset" "default" {
 }
 
 
-# This creates a BigQuery table named "my_table" in the dataset "default".
-# The table has three columns, named "country", "product", and "price", which are of types STRING, STRING, and INT64 respectively.
+# This creates a BigQuery Table wiht Automatic Metadata Caching.
```
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:84]:
```suggestion
# This creates a BigQuery Table with automatic metadata caching.
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf [hunk 12]
```diff
@@ -99,7 +98,17 @@ resource "google_bigquery_table" "default" {
     source_format = "PARQUET"
     connection_id = google_bigquery_connection.default.name
     source_uris   = ["gs://${google_storage_bucket.default.name}/data/*"]
+    # This enables automatic metadata refresh.
+    metadata_cache_mode = "AUTOMATIC"
   }
+
+  # `max_staleness` is required for automatic metadata caching and must be
+  # specified as an interval literal.
+  # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#interval_literals
```
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:107]:
```suggestion
# This sets the maximum staleness of the metadata cache to 10 hours.
```

**Review by adhiggs — CHANGES_REQUESTED** [review]

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 13]
```diff
@@ -44,10 +46,10 @@ resource "google_storage_bucket_object" "default" {
 }
 
 # This queries the provider for project information.
-data "google_project" "project" {}
+data "google_project" "default" {}
 
-# This creates a connection in the US region named "my-connection".
-# This connection is used to access the bucket.
+# This creates a connection in the US region named "my-connection". This
+# connection is used to access the bucket.
```
[the remark this answers is not in the document]
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:52]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf [hunk 14]
```diff
@@ -87,9 +89,8 @@ resource "google_bigquery_dataset" "default" {
   }
 }
 
-# This creates a BigQuery table named "my_table" in the dataset "default".
-# The table has a single column named "column1", which is of type STRING
-# and is nullable.
+# This creates a BigQuery Table with Partitioning and Automatic Metadata
+# Caching.
```
[the remark this answers is not in the document]
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:93]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf (hunk 10 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:120]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 11 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:84]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf (hunk 12 above)
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:107]: done.

**Review by msampathkumar — APPROVED** [review]

**Pull request merged**